ctms/app.py

```diff
@@ -275,22 +280,51 @@ def get_api_client(
         token,
         secret_key=token_settings["secret_key"],
     )
+    log_context = request.state.log_context
+    log_context["client_allowed"] = False
+
     if name is None:
+        log_context["auth_fail"] = "No or bad token"
         raise credentials_exception
+
+    log_context["client_id"] = name
     if namespace != "api_client":
+        log_context["auth_fail"] = "Bad namespace"
         raise credentials_exception
+
     api_client = get_api_client_by_id(db, name)
     if not api_client:
+        log_context["auth_fail"] = "No client record"
         raise credentials_exception
+
     return api_client
 
 
-def get_enabled_api_client(api_client: ApiClientSchema = Depends(get_api_client)):
+def get_enabled_api_client(
+    request: Request, api_client: ApiClientSchema = Depends(get_api_client)
+):
+    log_context = request.state.log_context
     if not api_client.enabled:
+        log_context["auth_fail"] = "Client disabled"
         raise HTTPException(status_code=400, detail="API Client has been disabled")
+    log_context["client_allowed"] = True
     return api_client
 
 
+@app.middleware("http")
+async def log_request(request: Request, call_next):
```

Comment:
Not sure if we want to, but I'm guessing this is where we could log the body in an encoded fashion if we wanted to log it to reproduce the issue at a later time... or throw it in a DLQ/Table to be stored for a later test? 

Not really thinking too much on this as it might not be a major concern...